File: frontend/src/app/shared/models/EvaluatorProfileModel/ResponseEvaluatorProfileDto.ts
import { SaveDiseaseExperienceDto } from './SaveDiseaseExperienceDto';

export interface ResponseEvaluatorProfileDto {
  id: number;
  professions: string[];
  sectors: string[];
  interventionLevels: string[];
  sectorsWorkedIn: string[];
  countries: string;
  regions: string;
  divisions: string;
  rvfExperience: SaveDiseaseExperienceDto;
  evdExperience: SaveDiseaseExperienceDto;
}

File: backend/src/main/java/com/webgis/evaluatorprofile/dto/ResponseEvaluatorProfileDto.java
package com.webgis.evaluatorprofile.dto;

import com.webgis.evaluatorprofile.EvaluatorProfile;

import java.util.List;

public record ResponseEvaluatorProfileDto(
        Long id,
        List<String> professions,
        List<String> sectors,
        List<String> interventionLevels,
        List<String> sectorsWorkedIn,
        String countries,
        String regions,
        String divisions,
        ResponseDiseaseExperienceDto rvfExperience,
        ResponseDiseaseExperienceDto evdExperience
) {
    public ResponseEvaluatorProfileDto(EvaluatorProfile evaluatorProfile) {
        this(
                evaluatorProfile.getId(),
                evaluatorProfile.getProfessions(),
                evaluatorProfile.getSectors(),
                evaluatorProfile.getInterventionLevels(),
                evaluatorProfile.getSectorsWorkedIn(),
                evaluatorProfile.getCountries(),
                evaluatorProfile.getRegions(),
                evaluatorProfile.getDivisions(),
                new ResponseDiseaseExperienceDto(evaluatorProfile.getRvfExperience()),
                new ResponseDiseaseExperienceDto(evaluatorProfile.getEvdExperience())
        );
    }
}